tests/framework/cli/test_catalog.py

```diff
@@ -360,3 +383,35 @@ def test_bad_env(self, fake_project_cli, fake_metadata):
 
         assert result.exit_code
         assert "Unable to instantiate Kedro session" in result.output
+
+
+@pytest.mark.usefixtures(
+    "chdir_to_dummy_project", "fake_load_context", "mock_pipelines"
+)
+def test_list_catalog_factories(
```

Comment:
I think it would also be useful to show where an exact match will show up (should be at the top).

Reply:
E.g. a catalog entry without any placeholders.